pkg/minikube/download/image.go

```diff
@@ -0,0 +1,246 @@
+/*
+Copyright 2020 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package download
+
+import (
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/cheggaaa/pb/v3"
+	"github.com/google/go-containerregistry/pkg/name"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/daemon"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/google/go-containerregistry/pkg/v1/tarball"
+	"github.com/juju/mutex"
+	"github.com/pkg/errors"
+	"k8s.io/klog/v2"
+	"k8s.io/minikube/pkg/minikube/constants"
+	"k8s.io/minikube/pkg/minikube/localpath"
+	"k8s.io/minikube/pkg/util/lock"
+)
+
+var (
+	defaultPlatform = v1.Platform{
+		Architecture: runtime.GOARCH,
+		OS:           "linux",
+	}
+)
+
+// ImageExistsInCache if img exist in local cache directory
+func ImageExistsInCache(img string) bool {
+	f := filepath.Join(constants.KICCacheDir, path.Base(img)+".tar")
+	f = localpath.SanitizeCacheDir(f)
+
+	// Check if image exists locally
+	klog.Infof("Checking for %s in local cache directory", img)
+	if st, err := os.Stat(f); err == nil {
+		if st.Size() > 0 {
+			klog.Infof("Found %s in local cache directory, skipping pull", img)
+			return true
+		}
+	}
+	// Else, pull it
+	return false
+}
+
+var checkImageExistsInCache = ImageExistsInCache
+
+// ImageExistsInDaemon if img exist in local docker daemon
+func ImageExistsInDaemon(img string) bool {
```

Comment:
any func not used outside a package should be private (start with lower case)

Reply:
Done.